prow/tide/tide.go

```diff
@@ -103,16 +103,24 @@ func toSimpleState(s kube.ProwJobState) simpleState {
 	return noneState
 }
 
-func pickSmallestNumber(prs []pullRequest) pullRequest {
+func pickSmallestPassingNumber(prs []pullRequest) (bool, pullRequest) {
 	smallestNumber := -1
 	var smallestPR pullRequest
 	for _, pr := range prs {
-		if smallestNumber < 0 || int(pr.Number) < smallestNumber {
-			smallestNumber = int(pr.Number)
-			smallestPR = pr
+		if smallestNumber != -1 && int(pr.Number) >= smallestNumber {
+			continue
+		}
+		if len(pr.Commits.Nodes) < 1 {
+			continue
 		}
+		// TODO(spxtr): Check the actual statuses for individual jobs.
+		if string(pr.Commits.Nodes[0].Commit.Status.State) != "SUCCESS" {
+			continue
+		}
+		smallestNumber = int(pr.Number)
+		smallestPR = pr
 	}
-	return smallestPR
+	return smallestNumber > -1, smallestPR
 }
 
 // accumulate returns the supplied PRs sorted into three buckets based on their
```

Comment:
I was rereviewing this helper and I am wondering whether we need to take the PR headref into account when filtering through prowjobs.

Reply:
We already do, in `divideSubpool`.

Reply:
Now I see, thanks.